### ISSUE TITLE
onCreateOptionsMenu called after activity is finished

### ISSUE THREAD

**Issue by jasonphanley** [JakeWharton/ActionBarSherlock#459]
I noticed that if an activity under goes 2 config changes in a row, `onCreateOptionsMenu` is called for the activity created by the first  config change after it has finished. This can happen on a device with a slide out keyboard when going from landscape to portrait. There will be separate config changes for the keyboard hidden and rotation config changes and they will happen right after each other.

To fix this I added a check for config changes (`mActivity.getChangingConfigurations() == 0`) before this gets dispatched. The change is on line 980 of ActionBarSherlockCompat.

```
// Post the panel invalidate for later; avoid application onCreateOptionsMenu
// being called in the middle of onCreate or similar.
mDecor.post(new Runnable() {
    @Override
    public void run() {
        //Invalidate if the panel menu hasn't been created before this.
        if (!mActivity.isFinishing() && mActivity.getChangingConfigurations() == 0 && mMenu == null) {
            dispatchInvalidateOptionsMenu();
        }
    }
});
```

**Comment by JakeWharton** [comment]
Does this happen on ICS too?

**Comment by jasonphanley** [comment]
I can't get the ICS emulator to do 2 config changes in a row, so I can't test. There is an android bug report about this behavior: http://code.google.com/p/android/issues/detail?id=2423. I was seeing this in the 2.2 emulator and it may have been fixed/optimized on later platforms.

**Comment by JakeWharton** [comment]
Probably fixed this with 17a701065e2254fa48a343b0701e5cb98904bdc3 but will need to test before closing the issue.

**Comment by jasonphanley** [comment]
Tested your commit and it fixes the behavior for the original case I saw it in.
